### PR TITLE
Inspect PR changes to determine if subproject builds are needed.

### PR DESCRIPTION
**Pull request by alliepiper**
```
Subprojects: libcudacxx cub thrust

Dependencies:
  - libcudacxx -> cccl
  - cub -> cccl libcudacxx thrust
  - thrust -> cccl libcudacxx cub

Base SHA: e3758cfa45ebde29893a149933efbb7071fac56
HEAD SHA: a7f6ff444ab9fb38000d11ba3a3fef4f292b3df5

Dirty files:
  - .github/actions/compute-matrix/compute-matrix.sh
  - .github/workflows/pr.yml
  - ci/inspect_changes.sh

Modifications in project?
X - non-project files
  - libcudacxx
  - cub
  - thrust

Modifications in project or dependencies?
X - libcudacxx
X - cub
X - thrust

LIBCUDACXX_DIRTY=true
CUB_DIRTY=true
THRUST_DIRTY=true
```

### CODE REVIEW

**Comment by alliepiper** [comment]
The copy-pr-bot doesn't expose base SHA information. I've filled an issue with them to see if they can somehow expose that info to us.

File: ci/inspect_changes.sh [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+  cccl
+  libcudacxx
+  thrust
+)
+
+thrust_dependencies=(
+  cccl
+  libcudacxx
+  cub
+)
```
[the remark this answers is not in the document]
Comment (alliepiper) [reply ci/inspect_changes.sh:29]: Not currently, though this is an interesting idea.

[the remark this answers is not in the document]
Comment (miscco) [reply ci/inspect_changes.sh:38]: @robertmaynard Do you know of a way to use cmake for this?

**Review by miscco — APPROVED** [review]

File: ci/inspect_changes.sh [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
```
Comment (jrhemstad) [inline ci/inspect_changes.sh:38]: bash has associative arrays which can simplify this logic a bit. (And avoids the `eval` later in `add_dependencies()`.)

```suggestion
# ...and their dependencies:
declare -A dependencies=(
  [libcudacxx]="cccl"
  [cub]="cccl libcudacxx thrust"
  [thrust]="cccl libcudacxx cub"
)
```

**Review by jrhemstad — APPROVED** [review]
Approving since none of my comments were blocking.

File: ci/inspect_changes.sh [hunk 3]
```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+
+# Github action script to identify which subprojects are dirty in a PR
+
+set -u
+
+head_sha=$GITHUB_SHA
+base_sha=$BASE_SHA
```
[the remark this answers is not in the document]
Comment (alliepiper) [reply ci/inspect_changes.sh:8]: Done.

File: ci/inspect_changes.sh (hunk 2 above)
Comment (alliepiper) [reply ci/inspect_changes.sh:38]: Great idea!

**Review by miscco — APPROVED** [review]

**Pull request merged**